cpp/core/shuffle/Options.h

```diff
@@ -56,6 +57,7 @@ struct PartitionWriterOptions {
   int32_t compressionThreshold = kDefaultCompressionThreshold;
   arrow::Compression::type compressionType = arrow::Compression::LZ4_FRAME;
   CodecBackend codecBackend = CodecBackend::NONE;
+  int32_t compressionLevel = std::numeric_limits<int>::min();
```

Comment:
Is it possible to use `kUseDefaultCompressionLevel` as the default value here?

Reply:
changed to use kUseDefaultCompressionLevel